core/configuration/src/main/kotlin/guildsettings/GuildSettingsRepository.kt

```diff
@@ -18,8 +18,10 @@ package guildsettings
 import kotlinx.coroutines.flow.Flow
 
 // TODO write higher level functions to alter the DB and expose those
-interface GuildSettingsDatabase {
+interface GuildSettingsRepository {
```

Comment:
this isn't a repository, it's too generic to be :)

a repository would expose some specific functionality about guild settings but this is an abstraction to save and retrieve data in a map-like fashion

we would have to define some kind of "GuildSetting" object and provide ways to update it / listen to it if we would want to call this a repository, but I feel like it makes sense to add it on top of what now is a GuildSettingsDatabase

Reply:
I did consider naming it `GuildSettings`, but that would conflict with the generated DB class name unfortunately
I'm open to ideas, but I felt like calling the interface a database isn't necessarily true

Reply:
I'm for keeping the GuildSettingDatabase and building, on top of it, a GuildSettingRepository that expose read-write-monitor operation on some GuildSetting data class that uses the database under the hood

Reply:
I think creating an extra layer here is overkill, since our database class is generated by SQLDelight.

Either way, I'll revert the name change for this PR, we can come back to it later